packages/waku/src/lib/utils/merge-vite-config.ts

```diff
@@ -8,18 +8,38 @@ import {
 // Avoid vite finding the config itself, instead, we handle it ourselves with the config argument
 export async function mergeUserViteConfig(config: UserConfig) {
   const resolvedViteConfig = await resolveViteConfig({}, 'serve');
+  const filterPlugins = (plugin: PluginOption[]): PluginOption[] =>
+    plugin.flatMap((plugin) => {
+      if (Array.isArray(plugin)) {
+        return [filterPlugins(plugin)];
+      }
+      if (plugin && 'name' in plugin) {
+        if (resolvedViteConfig.plugins.some((p) => p.name === plugin.name)) {
+          return [];
+        }
+        return [plugin];
+      }
+      return [plugin];
+    });
 
   const mergedViteConfig = mergeViteConfig(
     {
       ...resolvedViteConfig,
       configFile: false,
-      // weird error around plugin duplication when removed
+      // FIXME weird error around plugin duplication when removed
       plugins: resolvedViteConfig.plugins.filter(
-        (plugin) => !plugin.name.startsWith('vite:'),
+        (plugin) =>
+          !['vite:css-post', 'vite:import-analysis', 'vite:json'].includes(
```

Comment:
For now, removing these three is working. But, we are far from confident.
We don't know why these causes issues, and why we need to remove.
Any insights are welcome. FIXME please.

Meanwhile, let's merge.